### DISCUSSION TITLE
Support for netstandard2.0?

### DISCUSSION THREAD

**Discussion by kayhantolga** [betalgo/openai#86] [Polls]
Would you like this library to support netstandard2.0?

As personally I do not like supporting older versions for a couple of reasons. 

You can not get the benefit of new versions of .net and other libraries.
You need to spend more time on development to do some basic tasks. I prefer to spend this time on something more useful. 
If more developer follows my pattern we will gain a habit to update our libraries to the latest all the time. So fewer developers will suffer from outdated versions.
It is **boring** and everyone should update their software to the latest version all the time(yes, I live with elves)

Also, it is fairly easy to add support for netstandard2.0. but if no one is interested to use it, I prefer to keep it simple. Please share if you have any use cases you have to use netstandard2.0.

**Comment by gotmike** [comment]
i agree. dotnet 6 -> 7 and beyond.

**Reply by gotmike** [reply]
caveat to this maybe is embedded devices, IoT or the like? sometimes those are slow to adapt and already in service...

**Comment by ricaun** [comment]
Unfortunately, I'm stuck in the `NET. Framework 4.8`. I can't control the version when designing a plugin for other software... 

Maybe is a bad idea to support `netstandard2.0`, I'm used to coding in an old version, so I don't miss the features in version `net6.0`. Technically by supporting `netstandard2.0` the code in the library gonna be downgraded, which could be annoying to maintain the code without the features in version `net6.0`.

Supporting `netstandard2.0` is simpler for older systems to implement OpenAI features.

I'm using OkGoDoIt implementation that uses `netstandard2.0` by default, only supporting Completions and the repo looks abandoned 😅
* https://github.com/OkGoDoIt/OpenAI-API-dotnet

**Reply by kayhantolga** [reply]
I am thinking about creating a new project in the same solution with .net standard 2.0 and copy-paste most of the codes. It may start with fewer features but at least it is better than nothing. we can maintain it as much as possible.

**Reply by gotmike** [reply]
i def see the advantage in expanding the capabilities to older projects.

if you have a specific use for it, by all means makes sense. i guess in my experience i've been better off upgrading my projects to later versions of dotnet than making new stuff backward-compatible.

also, i can help with that repo. i worked with the creator a while back and can prob jump back in if you need as well.

**Reply by ricaun** [reply]
I started playing with OpenAI API this week and I only tested Completions and Image (DALL·E), the rest I still need to figure out what is used for.

> also, i can help with that repo. i worked with the creator a while back and can prob jump back in if you need as well.

That would be great, I was able to implement Create Image in the OkGoDoIt library as an extension, and technically the dependencies fit better in my project, by default Newtonsoft.Json is used for serialization and stuff.
